packages/wallet/src/redux/fundingState/indirectFunding/state.ts

```diff
@@ -0,0 +1,102 @@
+import { SharedIndirectFundingState, INDIRECT_FUNDING } from '../shared/state';
+export { SharedIndirectFundingState, INDIRECT_FUNDING };
+
+// ChannelFundingStatus
+export const FUNDER_CHANNEL_DOES_NOT_EXIST = 'FUNDER_CHANNEL_DOES_NOT_EXIST';
+export const FUNDER_CHANNEL_EXISTS = 'FUNDER_CHANNEL_EXISTS';
+export const CHANNEL_FUNDED = 'CHANNEL_FUNDED';
+
+// Funding status
+export type ChannelFundingStatus =
```

Comment:
When you define this type like so, it becomes clear that these things are better served being an `enum` rather than a union type.

```
enum Foo {
  bar = 'BAR',
  baz = 'BAZ',
}

interface Bar {
  type: Foo.bar;
}

interface Baz {
  type: Foo.baz;
}

type Foobar = Bar | Baz;

function exhaustPossibilities(foobar: Foobar) {
  switch (foobar.type) {
    case Foo.bar:
      return 'bar';
    case Foo.baz:
      return 'baz';
    default:
      return unreachable(foobar);
  }
}

function missOnePossibility(foobar: Foobar) {
  switch (foobar.type) {
    case Foo.bar:
      return 'bar';
    default:
      return unreachable(foobar); // Argument of type 'Baz' is not assignable to parameter of type 'never'.
  }
}
```

Reply:
This definition of the `Foobar` type would not work:
```
interface Foobar {
  type: Foo
}
```
The compiler isn't aware that all possibilities are exhausted in the switch block in `exhaustPossibilities`.

But, I think this is ok, since we seem to actually want the following pattern:
- Define a "base state" interface of shared properties
- Create a few states that extend the base state
- Explicitly whitelist valid states as a union type of the defined states